Library/Homebrew/formula.rb

```diff
@@ -1318,18 +1318,25 @@ def verify_download_integrity(fn)
 
   # @private
   def run_test
-    old_home = ENV["HOME"]
-    build, self.build = self.build, Tab.for_formula(self)
-    mktemp do
-      @testpath = Pathname.pwd
-      ENV["HOME"] = @testpath
-      setup_home @testpath
-      test
+    preserve_files = ARGV.keep_files?
+    begin
+      old_home = ENV["HOME"]
+      build, self.build = self.build, Tab.for_formula(self)
+      mktemp("#{name}-", { :preserve => preserve_files }) do
```

Comment:
`{ :preserve => preserve_files }` should be `:preserve => preserve_files` instead. Also, if you used `ARGV.keep_files?` directly in place of `preserve_files`, there would be no need for a separate `begin`-`ensure`-`end` block and the diff would be more useful. (Just a suggestion.)